### DISCUSSION TITLE
Unpublished pages throws 500 error

### DISCUSSION THREAD

**Discussion by vindhya124** [PiranhaCMS/piranha.core#1915] [General] [answered]
Is there a way we can throw 404 (Not found) instead of 500 error when application is trying to access an unpublished page? 

Thanks in advance

**Comment by tidyui** [comment, marked as answer]
This is actually handled for Razor Pages here:

https://github.com/PiranhaCMS/piranha.core/blob/master/core/Piranha.AspNetCore/Models/SinglePage.cs#L62

But we might add logic for it earlier in the routing as well. Right now you can solve it in you controllers by returning NotFound when the model loader returns null.

Best regards

**Reply by tidyui** [reply]
Resolved in issue #1916. Will be released in next version
